ha/core/controllers/pcs/node_controller.py

```diff
@@ -152,6 +149,119 @@ def status(self, nodeids: list = None) -> dict:
+
+    def validate_node(self, node_id: str) -> bool:
+        """
+        Check node is a part of cluster
+        Args:
+            node_id ([str]): Private fqdn define in conf store.
+        Raises: ClusterManagerError
+        Returns: bool
+        """
+        _, node_list = self._get_offline_nodes()
+        if node_id in node_list:
+            return True
+        raise ClusterManagerError(f"nodeid = {node_id} is not a valid node id")
+
+    def _get_offline_nodes(self):
+        """
+        Get list of offline nodes ids.
+        """
+        cluster_status_xml = self._execute.run_cmd(const.GET_CLUSTER_STATUS)
+        # create element tree object
+        root = ET.fromstring(cluster_status_xml[0])
+        node_list = []
+        nodes_ids = []
+        # iterate news items
+        for item in root.findall('nodes'):
+            # iterate child elements of item
+            for child in item:
+                if child.attrib['online'] == 'false':
+                    nodes_ids.append(child.attrib['id'])
+                node_list.append(child.attrib['name'])
+        Log.info(f"List of offline node ids in cluster in sorted ascending order: {sorted(nodes_ids)}")
+        return sorted(nodes_ids), node_list
+
+    def create_health_event(self, nodeid: str, event_type: str) -> dict:
+        """
+        Create health event
+        Args:
+            nodeid (str): nodeid
+            event_type (str): event type will be offline, online, failed
+
+        Returns:
+            dict: Return dictionary of health event
+        """
+        key = self._system_health._prepare_key(const.COMPONENTS.NODE_MAP.value, node_id=nodeid)
+        node_map_val = self._health_manager.get_key(key)
+        if node_map_val is None:
+            raise ClusterManagerError("Failed to fetch node_map value")
+        node_map_dict = ast.literal_eval(node_map_val)
+
+        timestamp = str(int(time.time()))
+        event_id = timestamp + str(uuid.uuid4().hex)
+        initial_event = {
+            const.EVENT_ATTRIBUTES.EVENT_ID : event_id,
+            const.EVENT_ATTRIBUTES.EVENT_TYPE : event_type,
+            const.EVENT_ATTRIBUTES.SEVERITY : EVENT_SEVERITIES.WARNING.value,
+            const.EVENT_ATTRIBUTES.SITE_ID : node_map_dict[NODE_MAP_ATTRIBUTES.SITE_ID.value],
+            const.EVENT_ATTRIBUTES.RACK_ID : node_map_dict[NODE_MAP_ATTRIBUTES.RACK_ID.value],
+            const.EVENT_ATTRIBUTES.CLUSTER_ID : node_map_dict[NODE_MAP_ATTRIBUTES.CLUSTER_ID.value],
+            const.EVENT_ATTRIBUTES.STORAGESET_ID : node_map_dict[NODE_MAP_ATTRIBUTES.STORAGESET_ID.value],
+            const.EVENT_ATTRIBUTES.NODE_ID : nodeid,
+            const.EVENT_ATTRIBUTES.HOST_ID : node_map_dict[NODE_MAP_ATTRIBUTES.HOST_ID.value],
+            const.EVENT_ATTRIBUTES.RESOURCE_TYPE : CLUSTER_ELEMENTS.NODE.value,
+            const.EVENT_ATTRIBUTES.TIMESTAMP : timestamp,
+            const.EVENT_ATTRIBUTES.RESOURCE_ID : nodeid,
```

Comment:
please verify that RESOURCE_IS is same as nodeid here

Reply:
I guess, node is already verified at the beginning so this may not be required. Please check.

Reply:
checked, should be fine.